### DISCUSSION TITLE
Durable Option: alwaysReturnThis

### DISCUSSION THREAD

**Discussion by kwhitley** [kwhitley/itty-durable#17] [Polls]
In itty-durable, there's an option (I personally use it) to "alwaysReturnThis", meaning, any class method without an explicit `return` will just return the contents of the Durable itself, as defined by `toJSON()` (you can override).  This allows you to fire methods from the Worker and always get the state of the Durable itself as the response.  Anything that explicitly returns will be honored over this fallback behavior.

Would you want this behavior on by default?  Also, in the comments, what would you call it if not "alwaysReturnThis"?

```js
import { createDurable } from 'itty-durable'

// create the parent class
const Durable = createDurable({
  persist: false,
  alwaysReturnThis: true,
})

// extend it
export class Counter extends Durable {
  constructor(state, env) {
    super(state, env)
    
    this.value = 0
  }

  increment() {
    this.value++ // no return, so call would return { value }
  }

  clear() {
    this.value = 0 // no return, so call would return { value }
  }

  foo() {
    return 'bar' // explicitly return 'bar'
  }
}

```

**Comment by Wallacy** [comment]
I did not used that on my app because the perfomance penalty. My DO is very big, but the principal reason that i dont thing that is a good idea, is in your own example: clear() has no return, and my typescript annotation will sai that, but you are returning values on a function that i will never used without fight with my linter.

**Reply by kwhitley** [reply]
How would your linter even connect the two?  The function call itself doesn't return anything, it's the stub's call to a "matching" method that ends up returning the object's response.

**Reply by kwhitley** [reply]
Def see your point about avoiding large payloads, but then the question is:

Would most people have simpler/lighter DO uses, or follow more into your pattern of large data?  This is just the initial "sane" defaults (to serve the majority).  I ultimately want the minority to have to switch the setting, not the majority.

**Reply by Wallacy** [reply]
But how is that will actually help anyone? If the person wants to return the value, the person should write that... If the person create a function like increment/clear that does not return data, so no data should be returned.

You are trying to figure out what the person may want... but at end of the day the person already write what the intention.

And about the linter, I just declare my DO interface as be the same type as my DO class. The basic about the itty-durable is to call the DO class like a local class.

**Reply by kwhitley** [reply]
Ah cool, was just curious!

And regards to helping anyone... if someone (like myself) can benefit from an assumed behavior, why not allow that?  Others that want to more fully control their returns can do so as well.  One does not prevent the other!

**Reply by Wallacy** [reply]
Im not saying that behaviour is wrong or theres no use case, is just my humble opinion the defaults should be what the person write. For everyone that is using typescript I do think this may cause confusion because the person will have the interface diverging from the actual use.

Sorry if my question is too hash, busy day, sometimes is hard to express our thinking... I should write "But how is that will help anyone that did not want that feature explicit?", for those people i do think that can just set the `alwaysReturnThis` to `true`,
